pkg/ottl/ottlfuncs/func_trace_id_test.go

```diff
@@ -6,53 +6,135 @@ package ottlfuncs
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
 )
 
 func Test_traceID(t *testing.T) {
-	tests := []struct {
-		name  string
-		bytes []byte
-		want  pcommon.TraceID
-	}{
+	runIDSuccessTests(t, traceID[any], []idSuccessTestCase{
+		{
+			name:  "create trace id from 16 bytes",
+			value: []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16},
+			want:  pcommon.TraceID([16]byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}),
+		},
 		{
-			name:  "create trace id",
-			bytes: []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16},
+			name:  "create trace id from 32 hex chars",
+			value: []byte("0102030405060708090a0b0c0d0e0f10"),
 			want:  pcommon.TraceID([16]byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}),
 		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			exprFunc, err := traceID[any](tt.bytes)
-			require.NoError(t, err)
-			result, err := exprFunc(nil, nil)
-			require.NoError(t, err)
-			assert.Equal(t, tt.want, result)
-		})
-	}
+	})
 }
 
 func Test_traceID_validation(t *testing.T) {
-	tests := []struct {
-		name  string
-		bytes []byte
-	}{
+	runIDErrorTests(t, traceID[any], traceIDFuncName, []idErrorTestCase{
+		{
+			name:  "byte slice less than 16 (15)",
+			value: []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+			err:   errIDInvalidLength,
+		},
+		{
+			name:  "byte slice longer than 16 (17)",
+			value: []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17},
+			err:   errIDInvalidLength,
+		},
 		{
-			name:  "byte slice less than 16",
-			bytes: []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+			name:  "byte slice longer than 32 (33)",
+			value: []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33},
+			err:   errIDInvalidLength,
 		},
 		{
-			name:  "byte slice longer than 16",
-			bytes: []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17},
+			name:  "invalid hex string",
+			value: []byte("ZZ02030405060708090a0b0c0d0e0f10"),
+			err:   errIDHexDecode,
 		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			_, err := traceID[any](tt.bytes)
-			require.Error(t, err)
-			assert.ErrorContains(t, err, "traces ids must be 16 bytes")
-		})
-	}
+	})
+}
+
+func BenchmarkTraceID(b *testing.B) {
```

Comment:
Just a nit, feel free to ignore, but all benchmarks here are sharing the same code with different inputs, which works fine with the scenarios-based /table-driven approach:


```go
	benches := []struct {
		name    string
		value   []byte
		literal bool
	}{
		...
	}

	for _, bb := range benches {
		b.Run(bb.name, func(b *testing.B) {
			literalGetter, err := ottl.NewTestingLiteralGetter(bb.literal, makeIDGetter(bb.value))
			require.NoError(b, err)
			expr, err := traceID(literalGetter)
			require.NoError(b, err)
			b.ReportAllocs()
			b.ResetTimer()
			for b.Loop() {
				result, err := expr(b.Context(), nil)
				require.NoError(b, err)
			}
		})
	}
```